### PR TITLE
Add hint text for ref field

### LINKED ISSUE #92: Add some help text for git ref
For both git ref, it would be useful to have some small help text under the textbox showing what they are.

image

For the ref, I think it should say "Branch, Tag or Commit to use. `HEAD` will use the default branch"

### PR DESCRIPTION
**Pull request by oliverroick**
Resolves #92 

- Adds `hint` prop to `TextField` and the generic `Field` component. The hint is rendered only where there's no error on the field.
- Add `profile-option-control-hint` css class to style the hint
- Adapt spacing and font size of `profile-option-control-error` so it's in line with the new hint class

### CODE REVIEW

File: src/components/form/fields.jsx [hunk 1]
```diff
@@ -19,15 +19,17 @@ function validateField(value, validateConfig, touched) {
...
+function Field({ id, label, hint, children, error }) {
   return (
     <div className={`profile-option-container ${error ? "has-error" : ""}`}>
       <div className="profile-option-label-container">
         <label htmlFor={id}>{label}</label>
       </div>
       <div className="profile-option-control-container">
         {children}
-
+        {hint && !error && (
```
Comment (yuvipanda) [inline src/components/form/fields.jsx:30]: Why hide the hint if an error is present? Can we just keep the error above the hint?

Comment (oliverroick) [reply src/components/form/fields.jsx:30]: We can. It's pretty standard practise to do it this way but not written in stone. we can show both.

Comment (yuvipanda) [reply src/components/form/fields.jsx:30]: @oliverroick i'd love for us to show both, as I think the hint may help people fix errors!

Comment (oliverroick) [reply src/components/form/fields.jsx]: Ok done.

**Pull request merged**